libnmstate/nm/dns.py

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+from operator import itemgetter
+from operator import methodcaller
+
+from libnmstate import iplib
+from libnmstate.nm import nmclient
+from libnmstate.schema import Dns
+from libnmstate.nm import ipv4 as nm_ipv4
+from libnmstate.nm import ipv6 as nm_ipv6
+from libnmstate.error import NmstateInternalError
+
+
+DNS_DEFAULT_PRIORITY_VPN = 50
+DNS_DEFAULT_PRIORITY_OTHER = 100
+
+
+def get_running():
+    dns_state = {
+        Dns.NAME_SERVERS: [],
+        Dns.SEARCHES: []
+    }
+    client = nmclient.client()
+    dnss = client.get_dns_configuration()
+    dnss.sort(key=methodcaller('get_priority'))
+    for dns in dnss:
+        iface_name = dns.get_interface()
+        for ns in dns.get_nameservers():
+            if iplib.is_ipv6_link_local_addr(ns, 128):
```

Comment:
Would be nicer to use a variable/constant for the 128  - is it the prefix length?

Reply:
Now using `IPV6_ADDRESS_LENGTH`